packages/fluentui/docs/src/components/Sidebar/Sidebar.tsx

```diff
@@ -31,586 +32,566 @@ const behaviorMenu: ComponentMenuItem[] = require('../../behaviorMenu');
+    }
 
-    const treeItems = this.getTreeItems();
-    const withComponents = treeItems.concat(componentTreeSection);
-    return this.getSectionsWithPrototypeSectionIfApplicable(withComponents, prototypesTreeItems);
+    return sections.find(section => {
+      return section.items.some(item => item.title?.to.startsWith(newAt));
+    })?.id;
   };
 
-  render() {
-    const sidebarStyles: ICSSInJSStyle = {
-      background: '#201f1f',
-      width: `${this.props.width}px`,
-      position: 'fixed',
-      overflowY: 'scroll',
-      top: 0,
-      left: 0,
-      padding: 0,
-      height: '100%',
-      zIndex: 1000,
-    };
+  const getSectionsWithPrototypeSectionIfApplicable = React.useCallback((currentSections, allPrototypes) => {
+    let prototypes = process.env.NODE_ENV === 'production' ? _.filter(allPrototypes, { public: true }) : allPrototypes;
 
-    const logoStyles: ICSSInJSStyle = {
-      marginRight: '0.5rem',
-      width: '36px',
+    if (prototypes.length === 0) {
+      return currentSections;
+    }
+    prototypes = removePublicTags(prototypes);
+    const prototypeTreeSection = {
+      id: 'prototypes',
+      title: 'Prototypes',
+      items: prototypes,
     };
+    return currentSections.concat(prototypeTreeSection);
+  }, []);
 
-    const changeLogUrl: string = `${config.repoURL}/blob/master/packages/fluentui/CHANGELOG.md`;
-    const allSectionsWithoutSearchFilter = this.getSectionsWithoutSearchFilter();
-
-    const escapedQuery = _.escapeRegExp(this.state.query);
-    const regexQuery = new RegExp(`.*${escapedQuery}`, 'i');
-    const allSectionsWithPossibleEmptySections = _.map(
-      allSectionsWithoutSearchFilter,
-      (section: HierarchicalTreeItemProps) => {
-        return {
-          ...section,
-          items: _.filter(section.items as HierarchicalTreeItemProps[], item =>
-            regexQuery.test((item.title as HierarchicalTreeTitleProps).content as string),
-          ),
-        };
-      },
-    );
+  const treeItems = React.useMemo(
+    () => getSectionsWithPrototypeSectionIfApplicable(baseTreeItems, prototypesTreeItems),
+    [getSectionsWithPrototypeSectionIfApplicable],
+  );
 
-    let allSections = _.filter(
-      allSectionsWithPossibleEmptySections,
-      (section: HierarchicalTreeItemProps) => Array.isArray(section.items) && section.items.length > 0,
-    );
+  React.useEffect(() => {
+    document.addEventListener('keydown', handleDocumentKeyDown);
 
-    if (this.state.query !== '') {
-      allSections = _.map(allSections, (section: HierarchicalTreeItemProps) => {
-        return { ...section, open: true };
-      });
-    }
+    const at = props.location.pathname;
+    const id = findActiveCategoryId(at, treeItems);
+    setActiveItemIds(prev => (prev.includes(id) ? prev : [...prev, id]));
+    return () => {
+      document.removeEventListener('keydown', handleDocumentKeyDown);
+    };
+  }, [handleDocumentKeyDown, props.location.pathname, treeItems]);
```

Comment:
I suggest to split these things to be separate hooks

Reply:
Done